src/math/Euler.js

```diff
@@ -3,99 +3,88 @@ import { Vector3 } from './Vector3.js';
 import { Matrix4 } from './Matrix4.js';
 import { MathUtils } from './MathUtils.js';
 
-const _matrix = new Matrix4();
-const _quaternion = new Quaternion();
-
-function Euler( x = 0, y = 0, z = 0, order = Euler.DefaultOrder ) {
-
-	this._x = x;
-	this._y = y;
-	this._z = z;
-	this._order = order;
-
-}
+class Euler {
 
-Euler.RotationOrders = [ 'XYZ', 'YZX', 'ZXY', 'XZY', 'YXZ', 'ZYX' ];
+	constructor( x = 0, y = 0, z = 0, order = Euler.DefaultOrder ) {
 
-Euler.DefaultOrder = 'XYZ';
-
-Object.defineProperties( Euler.prototype, {
-
-	x: {
+		this._x = x;
+		this._y = y;
+		this._z = z;
+		this._order = order;
 
-		get: function () {
+	}
 
-			return this._x;
+	static get RotationOrders() {
 
-		},
+		return [ 'XYZ', 'YZX', 'ZXY', 'XZY', 'YXZ', 'ZYX' ];
 
-		set: function ( value ) {
+	}
 
-			this._x = value;
-			this._onChangeCallback();
+	static get DefaultOrder() {
```

Comment:
We should do a setter for this one too.

Reply:
@mrdoob 

Thank you for reviewing.
Sorry, I got it wrong, I thought `DefaultOrder` is read-only too.
I will fix it.

Before that, I have a question: How do we declare Public class fields?

The [readable-writable static value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields) can be declare like
```js
class MyClass {

  static xxx = 1;

  constructor( x ) {
    this.x = x
  }

}
```

However It seems Buble transpiler [does not support the syntax](https://buble.surge.sh/#class%20MyClass%20%7B%0A%20%20%0A%20%20static%20xxx%20%3D%201%3B%0A%0A%20%20constructor(%20x%20)%20%7B%0A%20%20%20%20this.x%20%3D%20x%0A%20%20%7D%0A%0A%7D), while [Babel support that](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAECyCeBhcVoG8BQ1pehALmPgJbDRgXQC80AjANwa7AD2AdgQE4Cuw-LnABTQAHtACU6XNnwALYhAB0YmiNwBfJuqA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ces2015%2Cstage-0%2Cenv&prettier=false&targets=&version=7.11.1&externalPlugins=)

Which option would be preferable in three.js (with Buble)?

```js
class MyClass {

  constructor( x ) {
    this.x = x
  }

}

MyClass.xxx = 1;
```
or
```js
let _xxx = 1;

class MyClass {

  static get xxx = () => {
    return _xxx;
  }

  static set xxx = ( value ) => {
    _xxx = value;
  }

  constructor( x ) {
    this.x = x
  }

}
```

or other?
Thank you in advance🙇‍♂️

maybe @DefinitelyMaybe or @ianpurvis could give a suggestion?

Reply:
> ```js
> class MyClass {
> 
>   constructor( x ) {
>     this.x = x
>   }
> 
> }
> 
> MyClass.xxx = 1;
> ```
 I think this style is nice for read/write 👍

Reply:
Thanks, I'll do that then!

Reply:
For future reference:
There are only three things in the current `r120` dev build where Buble is being used.
`Curves`, `Geometries` and `Materials`.
When Buble is no longer needed, i.e. after converting everything to es6 classes, we should have a look back into this
```javascript
class MyClass {

  static xxx = 1;

  constructor( x ) {
    this.x = x
  }

}
```

Reply:
nvm. I've got Buble's purpose backwards. Buble is our 'convert classes back to support IE' tool and output it as `build/three.js`.